### PR TITLE
allow easy configuration of single or double tapping

### PR DESCRIPTION
**Pull request by ladyada**
see below!

```
import time
from adafruit_circuitplayground.express import cpx

# Set up the accelerometer to detect tapping ('click')
cpx.detect_taps = 1

while True:
    if cpx.tapped:     # Look for a single tap
        print("Tapped", time.monotonic())
    time.sleep(0.01)

```

### CODE REVIEW

File: adafruit_circuitplayground/express.py [hunk 1]
```diff
@@ -119,28 +119,53 @@ def __init__(self):
         self._lis3dh.range = adafruit_lis3dh.RANGE_8_G
 
         # Initialise tap:
+        self._last_tap = False
+        self._detect_taps = 1
+        self.detect_taps = 1
```
[the remark this answers is not in the document]
Comment (ladyada) [reply adafruit_circuitplayground/express.py]: one is the setter (detect_taps) and one is the variable that is set by the setter (_detect_taps)

File: adafruit_circuitplayground/express.py [hunk 2]
```diff
@@ -119,28 +119,53 @@ def __init__(self):
...
+              print("Single Tap detected!")
+        """
+        return self._detect_taps
+
+    @detect_taps.setter
+    def detect_taps(self, value):
+        self._detect_taps = value
         try:
-            self._lis3dh.set_tap(2, 18, time_limit=4, time_latency=17, time_window=110)
+            self._lis3dh.set_tap(value, 18, time_limit=4, time_latency=17, time_window=110)
```
[the remark this answers is not in the document]
Comment (ladyada) [reply adafruit_circuitplayground/express.py]: we can tweak it for single/double, maybe set threshhold to 24 or so for single!

[the remark this answers is not in the document]
Comment (kattni) [reply adafruit_circuitplayground/express.py]: I had to set single-tap threshold to 70 with them separated before single-tap wasn't going off constantly. At 70 it misses a deliberate tap rarely, and still gets a few accidental taps, but it seemed like a happy medium for single-tap. Double-tap doesn't work at that threshold level.

File: adafruit_circuitplayground/express.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kattni) [reply adafruit_circuitplayground/express.py]: Ah thank you!

**Comment by kattni** [comment]
It looks like 80 is a good threshold. 

When you said "tweak it for single/double" were you saying tweak the number in this implementation or did you want me to clean up the `single_tap`/`double_tap` implementation I linked?

File: adafruit_circuitplayground/express.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ladyada) [reply adafruit_circuitplayground/express.py]: sure sounds good, yeah with 8G its hard to get good numbers!

**Comment by ladyada** [comment]
yah i mean checking if you want 1 or 2 at this line
https://github.com/adafruit/Adafruit_CircuitPython_CircuitPlayground/pull/26/commits/3a0c653614f5197e6a35f27709679411122cec6b
and then changing what you call self._lis3dh.set_tap with. i can commit late tonite when i get home, or you can take a go :)

**Comment by kattni** [comment]
I'm still curious though, why detect_taps and tapped, rather than single_tap and double_tap?

**Comment by kattni** [comment]
I'll commit it after the weekly :)

**Comment by ladyada** [comment]
zoOOOOOoooM!

**Comment by kattni** [comment]
It doesn't look much different on the code end. The library sets everything up and the code is basically:
```
if cpx.double_tap:
    blah
elif cpx.single_tap:
    etc
```
Even with detect_taps and tapped, you can do both, but it looks more like this:
```
cpx.detect_taps = 2
if cpx.tapped:
    blah
cpx.detect_taps = 1
if cpx.tapped:
    etc
```
Either way, double doesn't really work with single. If you try to do both in LIS3DH, it fails, you can't create two instances of `lis3dh.set_tap`. It's not really designed to have two instances of it, and regardless of how we're implementing it in `cpx`, you can create two instances of it in your code.

Do we want to try to put in something that makes it fail if you try to use it twice? Or deal with it in documentation? Or something different?

**Comment by kattni** [comment]
I'd like to do some work on the documentation for these changes, update the explanations and add another example where you can delay between checking for single and double taps.

I went straight to trying to use both in a tight loop, so I assume others might too. So I thought if we explain more clearly that it's single OR double, and also have an example of being able to use both with a delay between checking for single and double, that we'll catch that well enough.

Let me know if this works for you and I'll make the changes!

**Comment by kattni** [comment]
I'm going to merge this PR and do the next set of updates locally.

**Pull request merged**